xapian-applications/omega/omindexcheck.cc

```diff
@@ -0,0 +1,106 @@
+/** @file omindexcheck.cc
+ * @brief Auxiliar program of omindextest
+ */
+/* Copyright (C) 2019 Bruno Baruffaldi
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+#include <config.h>
+#include "worker.h"
+
+#include <xapian.h>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <cstdlib>
+
+#include "gnu_getopt.h"
+#include "hashterm.h"
+#include "common/stringutils.h"
+
+using namespace std;
+
+typedef vector<string> testcase;
+
+unordered_map<string, testcase> tests;
+
+static void
+index_test(void)
+{
+    tests.insert({"test.txt", {"Zjoey", "Zfood",
+		  "Zедой", "Z喬伊不分享食物"}});
+#if defined HAVE_POPPLER
+    tests.insert({"pdf/hello.pdf", {"Shello", "Sworld", "Ajeroen", "Aooms",
+		  "Zhello", "Zworld", "Zsubsect"}});
+    tests.insert({"pdf/poppler.pdf", {"Zabstract", "Zkeyword", "Ztopic",
+		  "Ztext", "Spdftitle"}});
+#endif
+}
```

Comment:
I know that this test cases are poor. I just want to ask if this is way of testing is okay.